docs/topics/multiplatform/multiplatform-connect-to-apis.md

```diff
@@ -1,380 +1,276 @@
-from platform to platform. That's why you should use `expect`/`actual` declarations for this class.
+### Expected and actual functions
 
-In the common module, declare the expected class `PlatformSocket()` with the `expect` keyword. Don't include any implementation code.
+Define an expected function returning a value of this interface and actual functions returning its subclasses:
 
 ```kotlin
-//Common
-internal expect class PlatformSocket(
-    url: String
-) {
-    fun openSocket(listener: PlatformSocketListener)
-    fun closeSocket(code: Int, reason: String)
-    fun sendMessage(msg: String)
-}
-interface PlatformSocketListener {
-    fun onOpen()
-    fun onFailure(t: Throwable)
-    fun onMessage(msg: String)
-    fun onClosing(code: Int, reason: String)
-    fun onClosed(code: Int, reason: String)
-}
+// In the commonMain source set:
+interface Platform
+
+expect fun platform(): Platform
 ```
 
-In each platform-specific module (iOS and Android), provide the actual implementation for the class `PlatformSocket()`
-expected in the common module. Use the `actual` keyword to mark the actual implementation.
+```kotlin
+// In the androidMain source set:
+class AndroidPlatform : Platform
 
-The following examples show the implementation of this for Android and iOS.
+actual fun platform() = AndroidPlatform()
+```
 
 ```kotlin
-//Android
-import okhttp3.OkHttpClient
-import okhttp3.Request
-import okhttp3.Response
-import okhttp3.WebSocket
-
-internal actual class PlatformSocket actual constructor(url: String) {
-    private val socketEndpoint = url
-    private var webSocket: WebSocket? = null
-    actual fun openSocket(listener: PlatformSocketListener) {
-        val socketRequest = Request.Builder().url(socketEndpoint).build()
-        val webClient = OkHttpClient().newBuilder().build()
-        webSocket = webClient.newWebSocket(
-            socketRequest,
-            object : okhttp3.WebSocketListener() {
-                override fun onOpen(webSocket: WebSocket, response: Response) = listener.onOpen()
-                override fun onFailure(webSocket: WebSocket, t: Throwable, response: Response?) = listener.onFailure(t)
-                override fun onMessage(webSocket: WebSocket, text: String) = listener.onMessage(text)
-                override fun onClosing(webSocket: WebSocket, code: Int, reason: String) = listener.onClosing(code, reason)
-                override fun onClosed(webSocket: WebSocket, code: Int, reason: String) = listener.onClosed(code, reason)
-            }
-        )
-    }
-    actual fun closeSocket(code: Int, reason: String) {
-        webSocket?.close(code, reason)
-        webSocket = null
-    }
-    actual fun sendMessage(msg: String) {
-        webSocket?.send(msg)
-    }
+// In the iosMain source set:
+class IOSPlatform : Platform
+
+actual fun platform() = IOSPlatform()
+```
+
+WHen you call `platform()` in the common code, it can work with an object of the `Platform` type. When you run this common
```

Comment:
```suggestion
When you call the `platform()` function in the common code, it can work with an object of the `Platform` type. When you run this common
```